### DISCUSSION TITLE
Limit number of visualisations on dashboard

### DISCUSSION THREAD

**Discussion by vishwa-ims** [RevealBi/Reveal.Sdk#39] [ASP.NET Core]
Hi
Just wanted to check if we can limit number of visualisations on dashboard ? Currently we can add number of visualisations on a dashboard.

**Comment by brianlagunas** [comment]
No, there is no API to limit the number of visualizations that can be created. That seems like an unusual requirement. Could you explain your scenario in more detail so we can better understand what you are trying to achieve?

**Reply by vishwa-ims** [reply]
While creating a dashboard user can create number of visualisations on same dashboard. To avoid load on database we want to restrict user to certain number of visualisation say for example 10 so user can only have upto 10 visualisations on single dashboard. If he wants to have more than 10 visualisation , he should create 2nd dashboards . Else currently a user can have thousands of visualisations on single dashboard.

**Reply by brianlagunas** [reply]
Are you experiencing load on your database? Are your users' creating dashboards with thousands of visualizations? Or are you just thinking of scenarios that might be possible?

**Reply by vishwa-ims** [reply]
@brianlagunas  Yes we are experiencing load on data when number of users are trying to access same dashboard. And I believe we had raised one ticket to your team couple of months back and we got a feedback that we should limit maximum 20 visualisations on single dashboard to avoid performance issue

**Reply by brianlagunas** [reply]
If you are experiencing a load on your database, then you should be looking at your database infrastructure and configuration. I don't think limiting the number of visualizations is the right approach to this issue.
